header_test.go

```diff
@@ -11,6 +11,31 @@ import (
 	"testing"
 )
 
+func TestResponseHeaderMultiLineValue(t *testing.T) {
+	s := "HTTP/1.1 200 OK\r\n" +
+		"EmptyValue1:\r\n" +
+		"Content-Type: foo/bar;\r\n\tnewline;\r\n another/newline\r\n" + // the '\t' will be kept, won't be removed
+		"Foo: Bar\r\n" +
+		"Multi-Line: one;\r\ntwo\r\n" + // a space won't be inserted between 'one' and 'two'
+		"Values: v1;\r\n v2;\r\n v3; v4\r\n" +
+		"\r\n"
+	expectContentType := "foo/bar;\tnewline; another/newline"
+	expectMultiLine := "one;two"
+	header := new(RequestHeader)
+	_, err := header.parse([]byte(s))
+	if err != nil {
+		t.Fatalf("parse headers with multi-line values failed, %s", err)
+	}
+	gotContentType := header.Peek("Content-Type")
+	if string(gotContentType) != expectContentType {
+		t.Fatalf("unexpected content-type: %q. Expecting %q", gotContentType, expectContentType)
+	}
+	gotMultiLine := header.Peek("Multi-Line")
+	if string(gotMultiLine) != expectMultiLine {
+		t.Fatalf("unexpected multi-line: %q. Expecting %q", gotMultiLine, expectMultiLine)
+	}
+}
```

Comment:
Can you add one more test here, something like https://play.golang.org/p/Q8IJwhKbqHI which compares the header values against how `net/http` parses them.

The request you have in this test doesn't seem to be valid according to `net/http`: https://play.golang.org/p/wDchgwOABAg